scripts/wi-kind-setup.sh

```diff
@@ -0,0 +1,152 @@
+        echo "Please login to Azure CLI using 'az login'"
+        exit 1
+    fi
+
+    # check if RESOURCE_GROUP and LOCATION are provided
+    if [ -z "${RESOURCE_GROUP:-}" ] || [ -z "${LOCATION:-}" ]; then
+        echo "RESOURCE_GROUP and LOCATION are required."
+        exit 1
+    fi
+}
+
+create_azure_blob_storage_account() {
+    if [ "$(az group exists --name "${RESOURCE_GROUP}" --output tsv)" == 'false' ]; then
+        echo "Creating resource group '${RESOURCE_GROUP}' in '${LOCATION}'"
+        az group create --name "${RESOURCE_GROUP}" --location "${LOCATION}" --output none --only-show-errors
+    fi
+
+    if ! az storage account show --name "${AZURE_STORAGE_ACCOUNT}" --resource-group "${RESOURCE_GROUP}" > /dev/null 2>&1; then
+        echo "Creating storage account '${AZURE_STORAGE_ACCOUNT}' in '${RESOURCE_GROUP}'"
+        az storage account create --resource-group "${RESOURCE_GROUP}" --name "${AZURE_STORAGE_ACCOUNT}" --allow-blob-public-access true --output none --only-show-errors
+    fi
+
+    if ! az storage container show --name "${AZURE_STORAGE_CONTAINER}" --account-name "${AZURE_STORAGE_ACCOUNT}" > /dev/null 2>&1; then
+        echo "Creating storage container '${AZURE_STORAGE_CONTAINER}' in '${AZURE_STORAGE_ACCOUNT}'"
+        az storage container create --name "${AZURE_STORAGE_CONTAINER}" --account-name "${AZURE_STORAGE_ACCOUNT}" --public-access blob --output none --only-show-errors
+    fi
+}
+
+upload_openid_docs(){    
+    cat <<EOF > "${SCRIPT_PATH}/openid-configuration.json"
+{
+  "issuer": "${SERVICE_ACCOUNT_ISSUER}",
+  "jwks_uri": "${SERVICE_ACCOUNT_ISSUER}openid/v1/jwks",
+  "response_types_supported": [
+    "id_token"
+  ],
+  "subject_types_supported": [
+    "public"
+  ],
+  "id_token_signing_alg_values_supported": [
+    "RS256"
+  ]
+}
+EOF
+
+    echo "Uploading openid-configuration document to '${AZURE_STORAGE_ACCOUNT}' storage account"
+    upload_to_blob "${AZURE_STORAGE_CONTAINER}" "${SCRIPT_PATH}/openid-configuration.json" ".well-known/openid-configuration"
+
+    echo "Getting public signing key from the cluster"
+    kubectl get --raw /openid/v1/jwks | jq > "${SCRIPT_PATH}/jwks.json"
+    echo "Uploading jwks document to '${AZURE_STORAGE_ACCOUNT}' storage account"
+    upload_to_blob "${AZURE_STORAGE_CONTAINER}" "${SCRIPT_PATH}/jwks.json" "openid/v1/jwks"
+}
+
+upload_to_blob() {
+    local container_name=$1
+    local file_path=$2
+    local blob_name=$3
+
+    echo "Uploading ${file_path} to '${AZURE_STORAGE_ACCOUNT}' storage account"
+    az storage blob upload \
+        --container-name "${container_name}" \
+        --file "${file_path}" \
+        --name "${blob_name}" \
+        --account-name "${AZURE_STORAGE_ACCOUNT}" \
+        --output none --only-show-errors
+}
+
+create_kind_cluster() {
+    if [ "${SKIP_CLUSTER:-}" = "true" ]; then
+        echo "Skipping cluster creation"
+        return
+    fi
+
+    echo "Creating kind cluster"
+    KIND_CLUSTER_NAME="azure-workload-identity"
```

Comment:
Maybe we should define `DEFAULT_KIND_CLUSTER_NAME=azure-workload-identity` and `DEFAULT_KIND_IMAGE_VERSION=v1.29.0` near the top of the file. Then we can just refer to those values here and in the `help()` function.